src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -183,9 +183,16 @@ private async Task<bool> StopProjectAsync(HotReloadState hotReloadState, Cancell
 
             try
             {
-                await hotReloadState.Session.StopSessionAsync(default);
+                if (ImmutableInterlocked.TryRemove(ref _activeSessions, hotReloadState.Process.Id, out _))
```

Comment:
This performing a long running operation within a lock - should this be outside of the lock to avoid contention?

Reply:
Which lock is that?